app/views/questionnaire/index.html.haml

```diff
@@ -0,0 +1,13 @@
+%h1 Questionnaire
+%h2 #{@response}
+%hr
+- @display.each do |statement|
+  %h2 #{statement[:question]}
+  - statement[:answer].each do |a|
+    =link_to "#{a.ans}", user_questionnaire_index_path(:ans => a.id), :class => ("selected" if (@selected_a_id.include?(a.id)))
+    &nbsp 
+  %hr
+\
+\
+-if @end_of_questionnaire
+  =link_to 'Submit Questionnaire', user_questionnaire_index_path(:form => @selected_a_id[0])
```

Comment:
Could you also link back the submit to the user profile? and make a link to the questionairre from the user profile?